### ISSUE TITLE
[Build] Error in gs_alreadySetSwapInterval (wxGTK)

### ISSUE THREAD

**Issue by meerfrau** [wxWidgets/wxWidgets#24076]
### Build System Used

- [X] CMake
- [ ] configure
- [ ] mingw32-make with makefile.gcc
- [ ] MSBuild (Microsoft Visual Studio solution file)
- [ ] nmake with makefile.vc
- [ ] Xcode

```
  cmake -B build-gtk3 -S wxWidgets-$pkgver \
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DCMAKE_BUILD_TYPE=None \
    -DwxBUILD_TOOLKIT=gtk3 \
    -DwxUSE_OPENGL=ON \
    -DwxUSE_REGEX=sys\
    -DwxUSE_ZLIB=sys \
    -DwxUSE_EXPAT=sys \
    -DwxUSE_LIBJPEG=sys \
    -DwxUSE_LIBPNG=sys \
    -DwxUSE_LIBTIFF=sys \
    -DwxUSE_LIBLZMA=sys \
    -DwxUSE_PRIVATE_FONTS=ON \
    -DwxUSE_GTKPRINT=ON
  cmake --build build-gtk3
```

### Description
```
[ 87%] Building CXX object libs/gl/CMakeFiles/wxgl.dir/__/__/__/__/src/unix/glegl.cpp.o
./wxWidgets-3.2.4/src/unix/glegl.cpp: In member function 'virtual bool wxGLCanvasEGL::SwapBuffers()':
./wxWidgets-3.2.4/src/unix/glegl.cpp:813:11: error: 'gs_alreadySetSwapInterval' was not declared in this scope
  813 |     if ( !gs_alreadySetSwapInterval.count(this) )
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [libs/gl/CMakeFiles/wxgl.dir/build.make:141: libs/gl/CMakeFiles/wxgl.dir/__/__/__/__/src/unix/glegl.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:794: libs/gl/CMakeFiles/wxgl.dir/all] Error 2

```

### Platform and version information

- wxWidgets version  you are building: 3.2.4
- wxWidgets port  you are building: wxGTK
- OS  and its version: ArchLinux
- Compiler  being used: gcc 13.2.1
- Non-default compiler options, if any :

No Wayland, only X11!

**Comment by vadz** [comment]
Oops, sorry, we should definitely define `gs_alreadySetSwapInterval` outside of `GDK_WINDOWING_WAYLAND` check. I'll try to fix this a.s.a.p.

**Comment by UnitedMarsupials** [comment]
> Oops, sorry, we should definitely define `gs_alreadySetSwapInterval` outside of `GDK_WINDOWING_WAYLAND` check. I'll try to fix this a.s.a.p.

Or, maybe, references to it should be `GDK_WINDOWING_WAYLAND`-only? The section of `wxGLCanvasEGL::SwapBuffers()`, which refers to the global variable, seems to indicate in the comments, that it is Wayland-only too.

Separately, in any case, the variable should be made `static`, should it not?

**Comment by vadz** [comment]
> > we should definitely define `gs_alreadySetSwapInterval` outside of `GDK_WINDOWING_WAYLAND` check. I'll try to fix this a.s.a.p.

Just in case this went unnoticed, I did fix it in the commit referenced above and I plan to merge it soon.

> Or, maybe, references to it should be `GDK_WINDOWING_WAYLAND`-only?

No, it's really needed for X too.

> Separately, in any case, the variable should be made `static`, should it not?

No, it shouldn't, it's inside an anonymous namespace and so already has static linkage.

> Here is [the patch proposed to the FreeBSD port](https://bugs.freebsd.org/bugzilla/attachment.cgi?id=246724&action=diff) of [x11-toolkits/wxgtk32](https://FreshPorts.org/x11-toolkits/wxgtk32)

Sorry, but I think my commit above is better, please use it instead.

**Comment by vadz** [comment]
This was fixed by b8dc0eeb2fb6c49ae0ac9cbc86bd20c0bf3bdee7 some time ago, but I forgot to close this issue.
